tests/std/tests/GH_001964_constexpr_generate_canonical/test.cpp

```diff
@@ -0,0 +1,58 @@
+#include <cassert>
+#include <cmath>
+#include <cstdint>
+#include <random>
+#include <utility>
+
+using namespace std;
+
+int naive_iterations(const int bits, const uint64_t gmin, const uint64_t gmax) {
+    // Naive implementation of [rand.util.canonical]. Note that for large values of range, it's possible that
+    // log2(range) == bits when range < 2^bits. This can lead to incorrect results, so we can't use this function as
+    // a reference for all values.
+
+    const double range = static_cast<double>(gmax) - static_cast<double>(gmin) + 1.0;
+    return static_cast<int>(ceil(static_cast<double>(bits) / log2(static_cast<double>(range))));
+}
+
+void test(const int target_bits) {
+    // Increase the range until the number of iterations repeats.
+    uint64_t range = 2;
+    int k          = 0;
+    int prev_k     = -1;
+    while (k != prev_k) {
+        prev_k       = exchange(k, naive_iterations(target_bits, 1, range));
+        const int k1 = _Generate_canonical_iterations(target_bits, 1, range);

```

Comment:
We typically don't test internal machinery directly. While we _could_ test this indirectly via calling `generate_canonical` with URBGs that count the number of calls, I think it would be quite convoluted. If you think you could do so cleanly and readably then by all means do, but otherwise consider this an explanation to other reviewers why I am _not_ requesting a change here.

Reply:
What's your tolerance for `integer_sequence`s and precomputed tables? https://godbolt.org/z/jT9oeb7ns is what I've come up with.

Reply:
This is a good example of what I meant by "quite convoluted." I prefer the whitebox test we have in the code now.